opensoundscape/audiomoth.py

```diff
@@ -115,3 +114,59 @@ def parse_audiomoth_metadata_from_path(file_path):
             )
         else:
             return parse_audiomoth_metadata(metadata)
+
+
+def _parse_audiomoth_comment_dt(comment):
+    """parses start times as written in metadata Comment field of AudioMoths
+
+    examples of Comment Field date-times:
+    19:22:55 14/12/2020 (UTC-5)
+    10:00:00 15/05/2021 (UTC)
+
+    note that UTC-5 is not parseable by datetime, hence custom parsing
+    also note day-month-year format of date
+
+    Args:
+        comment: the full comment string from an audiomoth metadata Comment field
+    Returns:
+        localized datetime object in timezone specified by original metadata
+    """
+    # extract relevant portion of comment
+    dt_str = comment.split("Recorded at ")[1].split(" by ")[0]
+
+    # handle formats like "UTC-5" or "UTC+0130"
+    if "UTC-" in dt_str or "UTC+" in dt_str:
+        marker = "UTC-" if "UTC-" in dt_str else "UTC+"
+        dt_str_utc_offset = dt_str.split(marker)[1][:-1]
+        if len(dt_str_utc_offset) <= 2:
+            dt_str_tz_str = f"{marker}{int(dt_str_utc_offset):02n}00"
+        else:
+            dt_str_tz_str = f"{marker}{int(dt_str_utc_offset):04n}"
+
+        dt_str = f"{dt_str.split(marker)[0]}{dt_str_tz_str})"
+    else:  #
+        dt_str = dt_str.replace("(UTC)", "(UTC-0000)")
+    dt = datetime.datetime.strptime(
+        dt_str, "%H:%M:%S %d/%m/%Y (%Z%z)"
+    )  # .astimezone(final_tz)
+    return dt
+
+
+def _parse_audiomoth_battery_info(comment):
```

Comment:
I don't see a test for _parse_audiomoth_battery_info. 
I see it's a function for internal use by parse_audiomoth_metadata, which does have tests written. Do we need to write tests for internal functions?

Reply:
it's tested implicitly by the tests on parse_audiomoth_metadata. I think that in general, it's ok to have higher level tests that implicitly test lower level things, as long as it would be obvious what is responsible for the failure if the test fails.